crowbar_framework/app/controllers/nodes_controller.rb

```diff
@@ -427,7 +427,7 @@ def get_node_and_network(node_name)
     @node = NodeObject.find_node_by_name(node_name) if @node.nil?
     if @node
       # If we're in discovery mode, then we have a temporary DHCP IP address.
-      if not ["discovering", "discovered", "hardware-installing", "hardware-installed"].include? @node.state
+      if !["discovering", "discovered"].include?(@node.state)
```

Comment:
Hrm, good question. And the answer is: "it depends":

 - first, when doing hardware-installing/hardware-installed outside of the discovery process (that can happen with the bios/raid barclamps), then yes, it just works fine. But arguable, not really a case that we're seeing for all practical purposes ;-)

 - in discovery, we have two things:

  - the network role is not used, so it won't set the IP address from the host range. So at first, it would seem that you are correct, but...

  - the dhcp configuration is actually updated, and on lease renewal, the node will get the allocated IP address.

All in all, we are in a gray area here, and it pretty much depends on the timing. That being said, I would argue that people generally won't want to connect to the discovered node at this point in time (they would connect before allocating it) and what we display in the UI here is consistent with what is put in the DNS config. Which is why I lean towards keeping this.

(question: should I put all this in the commit message?)